src/nwbinspector/utils.py

```diff
@@ -90,9 +90,11 @@ def is_regular_series(series: np.ndarray, tolerance_decimals: int = 9):
 def is_ascending_series(series: Union[h5py.Dataset, ArrayLike], nelems: Optional[int] = None):
     """General purpose function for determining if a series is monotonic increasing."""
     if isinstance(series, h5py.Dataset):
-        return np.all(np.diff(_cache_data_selection(data=series, selection=slice(nelems))) > 0)
+        differences = np.diff(_cache_data_selection(data=series, selection=slice(nelems)))
     else:
-        return np.all(np.diff(series[:nelems]) > 0)  # already in memory, no need to cache
+        differences = np.diff(series[:nelems])
+
+    return np.all(differences >= 0)  # already in memory, no need to cache
```

Comment:
`>= 0` is needed because a negative number resolves to `True`

Reply:
I don't understand, what is the test that would fail:

```
def test_is_ascending_series():
    assert is_ascending_series(series=[1, 1, 1])
    assert is_ascending_series(series=[1, 2, 3])
    assert not is_ascending_series(series=[1, 2, 1])
```

I think I am missing something.

Reply:
oh sorry, I didn't read this all the way through. I see now you have moved the `>=` test later